internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8s.go

```diff
@@ -1764,7 +1766,36 @@ func (k *Kubernetes) reApplyErdaHPARules(sg apistructs.ServiceGroup) (interface{
 
 		scaledObj := convertToKedaScaledObject(scaledObject)
 
-		err = k.scaledObject.Patch(scaledObj.Namespace, scaledObj.Name, scaledObj)
+		old := &kedav1alpha1.ScaledObject{}
+
+		err = k.k8sClient.CRClient.Get(context.Background(), client.ObjectKey{
+			Namespace: scaledObj.Namespace,
+			Name:      scaledObj.Name,
+		}, old)
+
+		if err != nil {
+			return nil, err
+		}
+
+		spec := types.ScaledPatchStruct{}
+
+		spec.Spec.ScaleTargetRef = scaledObj.Spec.ScaleTargetRef
+		if *scaledObj.Spec.MinReplicaCount > 0 && *scaledObj.Spec.MaxReplicaCount > 0 && *scaledObj.Spec.MinReplicaCount < *scaledObj.Spec.MaxReplicaCount {
```

Comment:
Should check `scaleObj` is not nil before `*` at client side even if func `convertToKedaScaledObject` always return non-nil value.